packages/netlify-cms-core/src/actions/editorialWorkflow.js

```diff
@@ -242,6 +243,12 @@ export function loadUnpublishedEntry(collection, slug) {
   return async (dispatch, getState) => {
     const state = getState();
     const backend = currentBackend(state.config);
+    const entriesLoaded = get(state.editorialWorkflow.toJS(), 'pages.ids', false);
+    //run possible unpublishedEntries migration
+    if (!entriesLoaded) {
+      const response = await backend.unpublishedEntries(state.collections).catch(() => false);
```

Comment:
For published entry that have a draft entry, let's run the migration code first before loading the draft.

Reply:
This can happen when opening an entry in edit mode directly without going through the workflow tab first? Am I correct?

Reply:
Yes, that's correct.